Build failure with haskell-src-exts 1.18

```
> /tmp/stackage-build8$ stack unpack hprotoc-2.4.0
Unpacked hprotoc-2.4.0 to /tmp/stackage-build8/hprotoc-2.4.0/
> /tmp/stackage-build8/hprotoc-2.4.0$ runghc -clear-package-db -global-package-db -package-db=/var/stackage/work/builds/nightly/pkgdb Setup configure --package-db=clear --package-db=global --package-db=/var/stackage/work/builds/nightly/pkgdb --libdir=/var/stackage/work/builds/nightly/lib --bindir=/var/stackage/work/builds/nightly/bin --datadir=/var/stackage/work/builds/nightly/share --libexecdir=/var/stackage/work/builds/nightly/libexec --sysconfdir=/var/stackage/work/builds/nightly/etc --docdir=/var/stackage/work/builds/nightly/doc/hprotoc-2.4.0 --htmldir=/var/stackage/work/builds/nightly/doc/hprotoc-2.4.0 --haddockdir=/var/stackage/work/builds/nightly/doc/hprotoc-2.4.0 --flags=
Configuring hprotoc-2.4.0...
> /tmp/stackage-build8/hprotoc-2.4.0$ runghc -clear-package-db -global-package-db -package-db=/var/stackage/work/builds/nightly/pkgdb Setup build
Building hprotoc-2.4.0...
Preprocessing library hprotoc-2.4.0...
[ 1 of 13] Compiling Text.ProtocolBuffers.ProtoCompile.Identifiers ( Text/ProtocolBuffers/ProtoCompile/Identifiers.hs, dist/build/Text/ProtocolBuffers/ProtoCompile/Identifiers.o )
[ 2 of 13] Compiling Text.ProtocolBuffers.ProtoCompile.BreakRecursion ( Text/ProtocolBuffers/ProtoCompile/BreakRecursion.hs, dist/build/Text/ProtocolBuffers/ProtoCompile/BreakRecursion.o )
[ 3 of 13] Compiling Text.ProtocolBuffers.ProtoCompile.Gen ( Text/ProtocolBuffers/ProtoCompile/Gen.hs, dist/build/Text/ProtocolBuffers/ProtoCompile/Gen.o )

Text/ProtocolBuffers/ProtoCompile/Gen.hs:77:8: error:
    Not in scope: type constructor or class ‘SrcLoc’
```

Thanks, I will look into this.

On Thu, Sep 15, 2016 at 11:20 AM, Michael Snoyman <notifications@github.com>
wrote:

> > /tmp/stackage-build8$ stack unpack hprotoc-2.4.0
> Unpacked hprotoc-2.4.0 to /tmp/stackage-build8/hprotoc-2.4.0/
> > /tmp/stackage-build8/hprotoc-2.4.0$ runghc -clear-package-db -global-package-db -package-db=/var/stackage/work/builds/nightly/pkgdb Setup configure --package-db=clear --package-db=global --package-db=/var/stackage/work/builds/nightly/pkgdb --libdir=/var/stackage/work/builds/nightly/lib --bindir=/var/stackage/work/builds/nightly/bin --datadir=/var/stackage/work/builds/nightly/share --libexecdir=/var/stackage/work/builds/nightly/libexec --sysconfdir=/var/stackage/work/builds/nightly/etc --docdir=/var/stackage/work/builds/nightly/doc/hprotoc-2.4.0 --htmldir=/var/stackage/work/builds/nightly/doc/hprotoc-2.4.0 --haddockdir=/var/stackage/work/builds/nightly/doc/hprotoc-2.4.0 --flags=
> Configuring hprotoc-2.4.0...
> > /tmp/stackage-build8/hprotoc-2.4.0$ runghc -clear-package-db -global-package-db -package-db=/var/stackage/work/builds/nightly/pkgdb Setup build
> Building hprotoc-2.4.0...
> Preprocessing library hprotoc-2.4.0...
> [ 1 of 13] Compiling Text.ProtocolBuffers.ProtoCompile.Identifiers ( Text/ProtocolBuffers/ProtoCompile/Identifiers.hs, dist/build/Text/ProtocolBuffers/ProtoCompile/Identifiers.o )
> [ 2 of 13] Compiling Text.ProtocolBuffers.ProtoCompile.BreakRecursion ( Text/ProtocolBuffers/ProtoCompile/BreakRecursion.hs, dist/build/Text/ProtocolBuffers/ProtoCompile/BreakRecursion.o )
> [ 3 of 13] Compiling Text.ProtocolBuffers.ProtoCompile.Gen ( Text/ProtocolBuffers/ProtoCompile/Gen.hs, dist/build/Text/ProtocolBuffers/ProtoCompile/Gen.o )
>
> Text/ProtocolBuffers/ProtoCompile/Gen.hs:77:8: error:
>     Not in scope: type constructor or class ‘SrcLoc’
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/k-bx/protocol-buffers/issues/45>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABXVB6R0ucOSW2dDGSEBQz3ATU5RlMZiks5qqRvggaJpZM4J9tZX>
> .
>

I'm trying to build hprotoc 2.4.2 on nixos. Since I'm using the package definition from the nixpkgs repo (which is based on what's in hackage), it's trying to use `>= 1.16.0` for `haskell-src-exts`. I get the following:

```
[ 6 of 13] Compiling Text.ProtocolBuffers.ProtoCompile.Gen ( Text/ProtocolBuffers/ProtoCompile/Gen.hs, dist/build/Text/ProtocolBuffers/ProtoCompile/Gen.o )

Text/ProtocolBuffers/ProtoCompile/Gen.hs:77:8: error:
    Not in scope: type constructor or class ‘SrcLoc’
```

I'm a haskell noob, but I guess my best option is to fork this repo and find a way to get nix to use my version with `haskell-src-exts >= 1.16.0 && < 1.18`.

I wish I had the skills to help fix the code, but could this be fixed with a basic `haskell-src-exts` version constraint?

Thanks for your work on this library. :)

--- a/hprotoc/hprotoc.cabal
+++ b/hprotoc/hprotoc.cabal
@@ -35,7 +35,7 @@ Executable hprotoc
                    containers,
                    directory >= 1.0.0.1,
                    filepath >= 1.1.0.0,
-                   haskell-src-exts >= 1.16.0,
+                   haskell-src-exts >= 1.16.0 && < 1.18,
                    mtl,
                    parsec,
                    utf8-string

